csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -82,6 +100,43 @@ class AllocationOrderInferencer : public IterVisitor {
   std::unordered_map<const TensorView*, AllocationOrder>& alloc_order_map_;
 };
 
+TensorView* AllocationOrderInferencer::resolveAllocationOrder(
+    const std::vector<Val*>& candidates) {
+  TensorView* src = nullptr;
+  size_t non_bc_high_water_mark = 0;
+
+  // helper utils to count the number of non broadcast iterdomain
+  auto countNonBroadcastID = [](const TensorView* tv) -> size_t {
+    return std::count_if(
+        tv->getMaybeRFactorDomain().begin(),
+        tv->getMaybeRFactorDomain().end(),
+        [&](auto ptr_id) { return !ptr_id->isBroadcast(); });
+  };
+
+  for (auto* val_ptr : candidates) {
+    auto* tv_ptr = dynamic_cast<TensorView*>(val_ptr);
```

Comment:
```suggestion
  for (auto* val : candidates) {
    auto* tv = dynamic_cast<TensorView*>(val);
```

`auto*` already indicates they are pointers, so we don't need `_ptr` in the name.